subprojects/gst-plugins-good/sys/v4l2/gstv4l2bufferpool.c

```diff
@@ -476,7 +477,16 @@ gst_v4l2_buffer_pool_alloc_buffer (GstBufferPool * bpool, GstBuffer ** buffer,
       group = gst_v4l2_allocator_alloc_userptr (pool->vallocator);
       break;
     case GST_V4L2_IO_DMABUF_IMPORT:
-      group = gst_v4l2_allocator_alloc_dmabufin (pool->vallocator);
+      if (V4L2_TYPE_IS_OUTPUT(obj->type))
+        group = gst_v4l2_allocator_alloc_dmabufin (pool->vallocator);
+      else {
+        gst_buffer_pool_acquire_buffer (pool->other_pool, &downstream_buffer, NULL);
+        group = gst_v4l2_allocator_alloc_dmabufin_capture (pool->vallocator, pool->allocator, downstream_buffer);
+        if (group == NULL) {
+          gst_buffer_unref (downstream_buffer);
+          GST_DEBUG ("fail to create buf");
```

Comment:
Ust GST_DEBUG_OBJECT(), so we see which pool is being traced. Improve your trace as its missing the dmabuf import for capture device context.

Reply:
modified.